101-azurefirewall-hub-spoke-demo/azuredeploy.json

```diff
@@ -84,7 +80,8 @@
         "routeTables_azfw_spokes_via_fw_name": "azfw-spokes-via-fw",
     
         "virtualNetworkGateways_azfw_hub_gw_name": "azfw-hub-gw",
-        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw"    
+        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw",
+        "virtualNetworkGateways_shared_key":"[uniquestring(subscription().subscriptionId)]"    
```

Comment:
what's the requirement for uniqueness here?  This will allow for only one per subscription...

Reply:
Wanted to create a unique key for S2S connection. Subscr id as a seed may not be the best idea, you're right. Changed it to use deployment().name as the seed.